catalog/core/tests/test_views.py

```diff
@@ -207,86 +266,75 @@ def test_contact_view(self):
         self.without_login_and_with_login_test(
             self.reverse(CONTACT_US_URL), before_status=200)
 
-    def test_contact_info_submit_without_timestamp_and_security_hash(self):
-        response = self.post(self.reverse(CONTACT_US_URL, query_parameters={'format': 'json'}),
-                             {'name': 'John Watson', 'email': 'john@watson.com',
-                              'message': 'Sherlock want to use this application.'})
-        self.assertTrue(400, response.status_code)
-
-    def test_contact_info_submit_with_invalid_timestamp(self):
+    @settings(max_examples=50)
+    @given(st.text(),st.text(),st.text())
+    def test_contact_info_with_randomized_valid_data(self,name,message,contact):
         url = self.reverse(
             CONTACT_US_URL, query_parameters={'format': 'json'})
         response = self.get(url)
         json_data = json.loads(response.data['json'])
         security_hash = json_data['security_hash']
-        timestamp = float(json_data['timestamp']) + 1
-
-        response = self.post(url, {'name': 'John Watson',
-                                   'email': 'john@watson.com',
-                                   'message': 'Sherlock want to use this application.',
-                                   'timestamp': timestamp,
-                                   'security_hash': security_hash,
-                                   'contact_number': ''})
-        self.assertTrue(400, response.status_code)
-
-    def test_contact_info_submit_with_valid_timestamp_and_invalid_security_hash(self):
-        url = self.reverse(
-            CONTACT_US_URL, query_parameters={'format': 'json'})
-        response = self.get(url)
-        json_data = json.loads(response.data['json'])
-        security_hash = json_data['security_hash'] + 'fake'
-        timestamp = json_data['timestamp']
-        time.sleep(5)
-
-        response = self.post(url, {'name': 'John Watson', 'email': 'john@watson.com',
-                                   'message': 'Sherlock want to use this application.',
-                                   'timestamp': timestamp,
-                                   'security_hash': security_hash,
-                                   'contact_number': ''})
-        self.assertTrue(400, response.status_code)
+        timestamp = float(json_data['timestamp'])
+        user = models(User).example()
```

Comment:
This should be in the given decorator as GENERATE_USER. See the comment below for detail.